### DISCUSSION TITLE
First prototype is ready

### DISCUSSION THREAD

**Discussion by krisc-informatica** [krisc-informatica/arduino-smart-bike-trainer#6] [General]
Finally, the first prototype is ready and functioning with the BKOOL training app!

Some, or a lot, finetuning has to be done to get it working completely, but the training app recognizes the trainer and by pedalling, the ride is recorded!

To make it work, I had to implement a power meter function because BKOOL uses the power output from the smart trainer to calculate the speed. The current version does a rough power calculation and this must be improved.

I also have to do more tests to make the resistance work seemlessly. But these are details, imho.

The board schema is available at https://easyeda.com/editor#id=|19cc3184d50e48a2a1a3aa17352470df|2a5bb0665475417e845671ad6f72ec7e

**Comment by luc-h77** [comment]
Hi Krisc,
awesome project and work you've done, I'm currently trying to get it work and would be very happy to help with the power measurement finetuning,
I'm a programmer but not at all experienced in electronics - currently I'm facing the issue that I'm receiving only one sync signal from the 23V pin, when I plug in the cable of the trainer, no continuous synd..
Do you have an idea what the reason could be?

Cheers, Lukas

**Comment by krisc-informatica** [comment]
Hello,

Any help to make it better is welcome. I have made some changes in Friday
that I have to push online. I had errors for the grade and wind speed where
negative values were wrong.

Can you send a picture of the connections?

Op za 21 aug. 2021 om 12:40 schreef luc-h77 ***@***.***>

> Hi Krisc,
> awesome project and work you've done, I'm currently trying to get it work
> and would be very happy to help with the power measurement finetuning,
> I'm a programmer but not at all experienced in electronics - currently I'm
> facing the issue that I'm receiving only one sync signal from the 23V pin,
> when I plug in the cable of the trainer, no continuous synd..
> Do you have an idea what the reason could be?
>
> Cheers, Lukas
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/krisc-informatica/arduino-smart-bike-trainer/discussions/6#discussioncomment-1215534>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF2S2HFKWKNQLWOVO6KLHHTT557DFANCNFSM5CFBTMJQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>
-- 
Met vriendelijke groeten, Best regards, Mit freundlichen Grüßen,

Kris Cardinaels



*krisc-informatica bv - **informatica-oplossing op maat*
Kloterstraat 12, 3950 Bocholt-Kaulille
(t) +32 486 86 32 26
BTW BE 0732.676.434, RPR Antwerpen, afdeling Tongeren

**Reply by luc-h77** [reply]
Hi krisc, after reconnecting all cables it is working now.
Just one observation, in order to control the trainer's resistance I had to decommented the statement in row 387:
doWait = true;

With this the hardware is sorted and start to create on a nodejs application to compare my crankset powermeter with the Tacx 1680 power calculation side by side and hope I'll find a reasonable formula.

**Reply by StFalagar** [reply]
A lot of this work has already been done by Wouter and the guys at FortiusANT.  Have a look at the manual and details here...  it would make sense to create new classes in Fortius-Ant to handle this setup... FortiusAnt will already convert the signal to Bluetooth/Ant+ for Zwift etc. to pick up.

https://github.com/WouterJD/FortiusANT/blob/master/doc/FortiusANTUserManual.pdf

**Reply by krisc-informatica** [reply]
> A lot of this work has already been done by Wouter and the guys at FortiusANT. Have a look at the manual and details here... it would make sense to create new classes in Fortius-Ant to handle this setup... FortiusAnt will already convert the signal to Bluetooth/Ant+ for Zwift etc. to pick up.
> 
> https://github.com/WouterJD/FortiusANT/blob/master/doc/FortiusANTUserManual.pdf

Hello, a lot is done, but the trainers compatible with FortiusAnt are newer and already have some kind of digital protocol. The one used in my project is older and only has the brake signal.

**Reply by StFalagar** [reply]
Yes you are correct....  I just think it could save a lot of time and effort if you have your funtionality working to collaborate with Wouter and add it as additional options to FortiusAnt.  I also have a Tacx Flow Eddy Current Brake trainer that I use with a Fortius headunit.  They already have the power curves etc. for that built into FortiusAnt.  I guess what I am saying is if you have the communication working, you could possibly benefit a lot from using stuff they already have.
